apps/prairielearn/src/components/LinkSharing.tsx

```diff
@@ -0,0 +1,87 @@
+import { html } from '@prairielearn/html';
+
+export function PublicLinkSharing({
+  publicLink,
+  SharingMessage,
+  publicLinkMessage,
+}: {
+  publicLink: string;
+  SharingMessage: string;
+  publicLinkMessage: string;
+}) {
+  return html`
+    <div class="mb-3">
+      <label for="publicLink">Public link</label>
+      <span class="input-group">
+        <input
+          type="text"
+          class="form-control"
+          id="publicLink"
+          name="publicLink"
+          value="${publicLink}"
+          disabled
+        />
+        <button
+          type="button"
+          class="btn btn-sm btn-outline-secondary btn-copy"
+          data-clipboard-text="${publicLink}"
+          aria-label="Copy public link"
+        >
+          <i class="far fa-clipboard"></i>
+        </button>
+        <button
+          type="button"
+          class="btn btn-sm btn-outline-secondary"
+          aria-label="Public Link QR Code"
+          data-bs-toggle="modal"
+          data-bs-target="#publicLinkModal"
+        >
+          <i class="fas fa-qrcode"></i>
+        </button>
+      </span>
+      <small class="form-text text-muted"> ${publicLinkMessage} </small>
+    </div>
+  `;
+}
+
+export function StudentLinkSharing({
```

Comment:
If you really wanted, this component is almost identical to the `PublicLinkSharing` component. You could have it take in a `mode = student|public`, uppercase the mode, and use the mode to reduce all the duplicated code.

Reply:
there is some discussion over here: https://github.com/PrairieLearn/PrairieLearn/issues/11983 about how these links actually maybe _should_ have more different behavior than one another, so that would be an argument for leaving them seperately since they will probably further diverge.